### PR TITLE
improve interact

### PR DESCRIPTION
**Pull request by jeaniewu**
there is no trello task for this but I found interact really annoying
I make it more reactive (greater interaction coverage) also make the input button more reactive in input Manager

### CODE REVIEW

File: Assets/Scripts/Controller2.cs [hunk 1]
```diff
@@ -108,20 +109,63 @@ void move(float moveHorizontal, float moveVertical)
...
-		RaycastHit2D[] hits = 
-			Physics2D.RaycastAll(this.transform.position, direction,1.5f, layer);
-		Debug.DrawRay (this.transform.position ,direction*1.5f, Color.green,0.2f);
+		}
+
+		if (interactables.Count == 0)
+			return null;
+
+		// most common interactable in the list is the closest one
+		return mostCommon(interactables.ToArray());
```
Comment (jayfriso) [inline Assets/Scripts/Controller2.cs]: Oh we we're just tracking which object showed up most, that's bad.

File: Assets/Scripts/Controller2.cs [hunk 2]
```diff
@@ -108,20 +109,63 @@ void move(float moveHorizontal, float moveVertical)
...
+	}
+
+	private GameObject mostCommon(GameObject[] interactables){
```
Comment (jayfriso) [inline Assets/Scripts/Controller2.cs]: User this instead to get the mostCommon element:

> var most = list.GroupBy(i=>i).OrderByDescending(grp=>grp.Count())
>       .Select(grp=>grp.Key).First();
> http://stackoverflow.com/questions/355945/find-the-most-occurring-number-in-a-listint

Im surprised that the C# list doesn't provide a most common element function by default. But then again Im blessed to work with scala at work.

**Comment by jayfriso** [comment]
Looks good besides that one comment... though I'm still a bit confused how this improves the interaction. Isn't the only difference here that we're getting the most common interactable object we find while before we we're calling interact on everything? Is it because we we're calling interact multiple times on the same object before?

**Comment by jeaniewu** [comment]
no I used to only make one line of raycast, so thats why sometime that line is not touching the object that we want. Now I'm making more raycasts with angle -45 to 45 degree, looks like a radar now

**Comment by jayfriso** [comment]
:+1:

**Pull request merged**